packages/devtools/src/useReduxDevToolsExtension.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import type {
+  NavigationContainerRef,
+  NavigationState,
+  NavigationAction,
+} from '@react-navigation/core';
+import deepEqual from 'deep-equal';
+
+type DevToolsConnection = {
+  init(value: any): void;
+  send(action: any, value: any): void;
+  subscribe(
+    listener: (message: { type: string; [key: string]: any }) => void
+  ): () => void;
+};
+
+type DevToolsExtension = {
+  connect(options: { name: string }): DevToolsConnection;
+  disconnect(): void;
+};
+
+declare const __REDUX_DEVTOOLS_EXTENSION__: DevToolsExtension | undefined;
+
+export default function useReduxDevToolsExtension(
+  ref: React.RefObject<NavigationContainerRef>
+) {
+  const devToolsRef = React.useRef<DevToolsConnection>();
+
+  if (
+    devToolsRef.current === undefined &&
+    typeof __REDUX_DEVTOOLS_EXTENSION__ !== 'undefined'
+  ) {
+    devToolsRef.current = __REDUX_DEVTOOLS_EXTENSION__.connect({
+      name: '@react-navigation/devtools',
+    });
+  }
+
+  const lastStateRef = React.useRef<NavigationState | undefined>();
+  const lastActionRef = React.useRef<NavigationAction | undefined>(undefined);
```

Comment:
Nit - why specify `undefined` explicitly here?

Reply:
Probably autocomplete added it by mistake xD